streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -248,12 +247,14 @@ public void testGlobalThreadCloseWithoutConnectingToBroker() {
         // make sure we have the global state thread running too
         builder.globalTable("anyTopic");
         final KafkaStreams streams = new KafkaStreams(builder.build(), props);
-        streams.start();
-        streams.close();
+        try {
+            streams.start();
+        } catch (final StreamsException expected) {
```

Comment:
If we expect this exception to happen, we should so a `fail` after `stream.start()`?

If we are afraid it might not timeout, but hang, should we put a timeout annotation to the test?

Reply:
Ah, I was unaware of that annotation. I thought they all just timeout and fail after 30s.

I didn't add the `fail` because it's not important to verify that we get the exception. We are only verifying that the call doesn't hang.

I do think that we'll get the exception every time, so I could add the `fail`, I just chose to test the behavior we *want* (non-hanging), not the tightest bound on the behavior we observe (the exception). WDYT?

Reply:
I think we should test for the exception -- if we change the behavior intentionally, we should remove the `fail` as well as the `try-catch` instead of allowing both behavior to pass (it's an either or form my point of view). IMHO, tests should be on an as narrow code path as possible: if we change behavior and a test fails, we are forces to reflect on the change, what is good as it guards against undesired behavior changes...